app/src/main/java/mainburg/planetenweg/LocationMarker.java

```diff
@@ -0,0 +1,146 @@
+import android.content.pm.PackageManager;
+import android.location.Location;
+import android.location.LocationListener;
+import android.location.LocationManager;
+import android.os.Bundle;
+import android.support.v4.app.ActivityCompat;
+import android.support.v4.content.ContextCompat;
+import android.widget.Toast;
+import com.google.android.gms.maps.CameraUpdateFactory;
+import com.google.android.gms.maps.GoogleMap;
+import com.google.android.gms.maps.model.LatLng;
+import com.google.android.gms.maps.model.Marker;
+import com.google.android.gms.maps.model.MarkerOptions;
+
+import java.util.Timer;
+import java.util.TimerTask;
+
+/**
+ * The LocationMarker places a marker on your current location and
+ * updates it after a certain interval.
+ * Created by Markus on 04.10.2017.
+ */
+public class LocationMarker implements LocationListener {
+
+    private GoogleMap map;
+    private final Activity activity;
+    private final LocationManager gps;
+    private Marker position;
+    static final int GPS_PERMISSION_REQUEST_CODE = 1227;
+
+    private boolean active;
+
+    /**
+     * This variable indicates whether or not a location has been founb.
+     * It is needed to notify the user when his location could not have been determined.
+     */
+    private boolean locationFound;
+
+    public LocationMarker(Activity activity) {
+        this.map = map;
+        this.activity = activity;
+        active = false;
+        gps = (LocationManager) activity.getSystemService(Context.LOCATION_SERVICE);
+    }
+
+    /**
+     * Enables the Location Marker
+     * @param map The GoogleMap on which to place the marker
+     */
+    public void enable(GoogleMap map) {
+        this.map = map;
+        active = true;
+        if (ContextCompat.checkSelfPermission(activity, android.Manifest.permission.ACCESS_FINE_LOCATION)
+                == PackageManager.PERMISSION_GRANTED) {
+            refresh();
+        } else {
+            ActivityCompat.requestPermissions(activity, new String[] {
+                    android.Manifest.permission.ACCESS_FINE_LOCATION
+            },GPS_PERMISSION_REQUEST_CODE);
+        }
+    }
+    /**
+     * Disables the Location Marker
+     */
+    public void disable() {
+        position.remove();
+        active = false;
+    }
+
+    public boolean isEnabled() {
+        return active;
+    }
+
+    /**
+     * Sends a new request to determine the user's location.
+     * Make sure you have the required GPS permissions before calling this method.
+     */
+    public void refresh() {
+        gps.requestLocationUpdates(
+                LocationManager.GPS_PROVIDER,
+                5000, //the minimum time interval for notifications, in milliseconds. This field is only used as a hint
+                //to conserve power, and actual time between location, updates may be greater or lesser than this value. (See https://androidexample.com/GPS_Basic__-__Android_Example/index.php?view=article_discription&aid=68&aaid=93)
```

Comment:
please add hard line breaks and add one space after a comment

```java
// like this.
```

Reply:
Btw, you can just use constants with a good identifier and pass those, that way you can save yourself the comments and it's easier to adjust parameters.